kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -410,13 +433,8 @@ public int buildTransactionsForTransferOfFundsGeneration(String documentNumber,
         
         while (budgetAdjustmentTransactions.hasNext()) {            
             try {
-                ba = (BudgetAdjustmentTransaction) budgetAdjustmentTransactions.next();                 
-                Account baAccount = ba.getAccount();
-                if (baAccount.getIncomeStreamFinancialCoaCode().equals(incomeString[0]) && 
-                        baAccount.getIncomeStreamAccountNumber().equals(incomeString[1]) &&
-                        ba.getUniversityFiscalYear().equals(new Integer(incomeString[2])) &&
-                        ba.getUniversityFiscalPeriodCode().equals(incomeString[3])) {
-                    
+                ba = (BudgetAdjustmentTransaction) budgetAdjustmentTransactions.next();
+                if (isRecordTransactionMatch(ba, incomeString)) {                   
```

Comment:
The extra method makes a difference in how easy is to understand what's going on...